### PR TITLE
linuxPackages.nvidia_x11_legacy390: use patches from archlinux

### PR DESCRIPTION
**Pull request by flokli**
Those seem to have gotten outdated, and don't build on current kernels
anymore.

Reported in https://github.com/NixOS/nixpkgs/issues/62012

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by flokli** [comment]
It'd be awesome if @vaibhavsagar or somebody else with the hardware could see if this works on both master, and if backported to 19.03.

**Comment by vaibhavsagar** [comment]
I don't know how to change my system `<nixpkgs>` to use this PR, but I can confirm that it correctly builds for me.

**Comment by flokli** [comment]
@vaibhavsagar While calling `nixos-rebuild switch`, you can pass a `-I nixpkgs=path/to/checkout` pointing to a checkout of the nixpkgs repo.

You probably want to checkout the `release-19.03` branch and `git cherry-pick c2cd3a94e2f1e42beb3203094e8fd261970c557c`, as this PR is against the master branch (and you probably don't want to switch your system to unstable)

**Comment by vcunat** [comment]
You might prefer `-I nixpkgs=https://github.com/nixos/nixpkgs/archive/c2cd3a9.tar.gz`

File: pkgs/os-specific/linux/nvidia-x11/default.nix [hunk 1]
```diff
@@ -41,7 +41,15 @@ rec {
...
-    patches = lib.optional (kernel.meta.branch == "4.19") ./drm_mode_connector.patch;
+    patches = [
+    (fetchurl {
+      url = "https://git.archlinux.org/svntogit/packages.git/plain/trunk/kernel-4.16.patch?h=2ad07241ea525a6b6b555b6cb96a97634a4b2cb0";
+      sha256 = "11b3dp0na496rn13v5q4k66bf61174800g36rcwj42r0xj9cfak2";
+    })
+    (fetchurl {
+      url = "https://git.archlinux.org/svntogit/packages.git/plain/trunk/kernel-5.1.patch?h=42d50ef8d6048608d18bdf2c296dd335260c5a1a";
+      sha256 = "03v46ym2bcckg9q2xrilkg21hfiwypr6gl4jmly2q3m4yza9ja6r";
+    })];
```
Comment (vcunat) [inline pkgs/os-specific/linux/nvidia-x11/default.nix]: The file should be deleted, right?

Comment (flokli) [reply pkgs/os-specific/linux/nvidia-x11/default.nix]: `pkgs/os-specific/linux/nvidia-x11/drm_mode_connector.patch` didn't apply cleanly anymore. I deleted it now and updated the commit.

I wanted to stick with what archlinux provides, as their patches use `#if LINUX_VERSION_CODE < KERNEL_VERSION(…)` and seem to apply on recent kernels.

**Comment by flokli** [comment]
Maybe `-I nixpkgs=https://github.com/flokli/nixpkgs/archive/linuxPackages.nvidia_x11_legacy390.tar.gz`

But as I said, this will switch the system to unstable (until next rebuild-switch), so manually cherry-picking might be more suitable.

**Comment by vcunat** [comment]
Hmm, right, I didn't notice 19.03 was wanted.  GitHub provides nice `git clone` paths for pull requests (by number), but I can't see similar archive URLs.

**Comment by vaibhavsagar** [comment]
I manually cherry-picked onto `release-19.03` and used that branch for `nixos-rebuild` and it works great!

**Comment by flokli** [comment]
Glad to hear! Merged in and backported to 19.03 in c8124c121522134a1b60ccfcfdd3768049d76098.

**Comment by vaibhavsagar** [comment]
Thanks so much!

**Pull request merged**